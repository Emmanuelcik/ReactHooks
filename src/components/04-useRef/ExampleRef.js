import React ,{useState} from 'react';
import {MultipleCustomHook} from "../03-examples/MultipleCustomHook";
import "../02-useEffect/effects.css";


export const ExampleRef = () => {

    const [show, setShow] = useState(false);

    return (
        <div>
            <h1>Example ref</h1>
            <hr/>

            {show && <MultipleCustomHook />}

            <button
                className="btn btn-primary mt-5"
                onClick={() => {
                    setShow(!show);
                }}
            >
                Example ref
            </button>
        </div>
    )
}
