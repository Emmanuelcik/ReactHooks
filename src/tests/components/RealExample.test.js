import "../../setupTests";
import {shallow} from "enzyme";
import {ExampleRef} from "../../components/04-useRef/ExampleRef";

describe("Testing the component ExampleRef", () => {
    
    const wrapper = shallow( <ExampleRef /> );

    test("Should match the snapshot", () => {

        expect( wrapper).toMatchSnapshot();
        expect( wrapper.find("MultipleCustomHook").exists()).toBe(false);
        
    });
    test("Should match the snapshot", () => {
        
        wrapper.find("button").simulate("click");
        expect( wrapper.find("MultipleCustomHook").exists()).toBe(true);
    });
})