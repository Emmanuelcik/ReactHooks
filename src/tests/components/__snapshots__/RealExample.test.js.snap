// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the component ExampleRef Should match the snapshot 1`] = `
<div>
  <h1>
    Example ref
  </h1>
  <hr />
  <button
    className="btn btn-primary mt-5"
    onClick={[Function]}
  >
    Example ref
  </button>
</div>
`;
